### ISSUE TITLE
Eloquent helper file generator creates files with multiple identical declarations

### ISSUE THREAD

**Issue by hesje** [laravel-idea/plugin#235]
When I run the Eloquen helper file generator (with mixins) it usually does generate  the file, but phpstorm complains about variables, attributes, etc. that are declared multiple times. Especially with the spatie/laravel-permissions and laravel's own notifications package.
I even tried completely reinitializing my project in phpstorm, but to no avail.
Am I overlooking smth or is this a bug?

```
com.intellij.psi.PsiInvalidElementAccessException: Element: class com.jetbrains.php.lang.psi.PhpFileImpl #PHP  because: file:///Users/hesselbierma/Documents/DuplexxDev/git/myinitio/vendor/spatie/laravel-permission/src/helpers.php (invalid) is invalid
invalidated at: see attachment
	at com.intellij.extapi.psi.StubBasedPsiElementBase.getNode(StubBasedPsiElementBase.java:121)
	at com.intellij.psi.PsiInvalidElementAccessException.getPsiInvalidationTrace(PsiInvalidElementAccessException.java:98)
	at com.intellij.psi.PsiInvalidElementAccessException.<init>(PsiInvalidElementAccessException.java:61)
	at com.intellij.psi.PsiInvalidElementAccessException.<init>(PsiInvalidElementAccessException.java:33)
	at com.intellij.psi.util.PsiUtilCore.ensureValid(PsiUtilCore.java:478)
	at com.intellij.psi.impl.source.resolve.ResolveCache.ensureValidPsi(ResolveCache.java:234)
	at com.intellij.psi.impl.source.resolve.ResolveCache.ensureValidResults(ResolveCache.java:227)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:204)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:149)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:164)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:157)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.multiResolve(PhpReferenceImpl.java:153)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.resolve(PhpReferenceImpl.java:139)
	at com.laravel_idea.plugin.psi.helpers.a.i.b(i.java:56)
	at com.laravel_idea.plugin.psi.helpers.a.i.a(i.java:17)
	at com.laravel_idea.plugin.psi.helpers.n.a(n.java:13)
	at com.laravel_idea.plugin.psi.helpers.n.a(n.java:12)
	at com.laravel_idea.plugin.eloquent.g.d.a(d.java:20)
	at com.laravel_idea.plugin.e.a.b$f.a(b$f.java:362)
	at com.laravel_idea.plugin.e.a.b$f.invoke(b$f.java:22)
	at com.laravel_idea.plugin.eloquent.g.c.visitElement(c.java:17)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
	at com.jetbrains.php.lang.psi.elements.impl.PhpASTElementImpl.accept(PhpASTElementImpl.java:29)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:67)
	at com.intellij.psi.PsiWalkingState.visit(PsiWalkingState.java:24)
	at com.intellij.util.WalkingState.walkChildren(WalkingState.java:62)
	at com.intellij.util.WalkingState.elementStarted(WalkingState.java:49)
	at com.intellij.psi.PsiWalkingState.elementStarted(PsiWalkingState.java:76)
	at com.intellij.psi.PsiRecursiveElementWalkingVisitor.visitElement(PsiRecursiveElementWalkingVisitor.java:48)
	at com.laravel_idea.plugin.eloquent.g.c.visitElement(c.java:24)
	at com.jetbrains.php.lang.psi.elements.impl.GroupStatementImpl.accept(GroupStatementImpl.java:25)
	at com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
	at com.laravel_idea.plugin.e.a.b.a(b.java:369)
	at com.laravel_idea.plugin.e.a.b.a(b.java:22)
	at com.laravel_idea.plugin.e.a.b$h.a(b$h.java:353)
	at com.laravel_idea.plugin.e.a.b$h.invoke(b$h.java:22)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.laravel_idea.plugin.e.a.b.b(b.java)
	at com.laravel_idea.plugin.e.a.b.a(b.java:82)
	at com.laravel_idea.plugin.e.a.b.a(b.java:42)
	at com.laravel_idea.plugin.e.a.a.a(a.java:78)
	at com.laravel_idea.plugin.generation.models.settings.specialGenerations.models.IdeHelperModelsGeneration$b$a.a(IdeHelperModelsGeneration$b$a.java:104)
	at com.laravel_idea.plugin.generation.models.settings.specialGenerations.models.IdeHelperModelsGeneration$b$a.compute(IdeHelperModelsGeneration$b$a.java:95)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:875)
	at com.laravel_idea.plugin.generation.models.settings.specialGenerations.models.IdeHelperModelsGeneration$b.a(IdeHelperModelsGeneration$b.java:103)
	at com.laravel_idea.plugin.generation.models.settings.specialGenerations.models.IdeHelperModelsGeneration$b.compute(IdeHelperModelsGeneration$b.java:95)
	at com.intellij.openapi.progress.Task$WithResult.run(Task.java:309)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:962)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:520)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:79)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:235)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:235)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by adelf** [comment]
Hi. Thanks for reporting. Could you tell me the Laravel Idea version?

**Comment by hesje** [comment]
The version is: 4.1.2.203

**Comment by adelf** [comment]
I've changed the algorithm in the 4.2 version. So, please update and make File > Invalidate caches and restart. It should help.

**Comment by adelf** [comment]
I've found a bug with generation. Thanks. 
It happens when for example User class has `HasRoles` and `HasPermissions` traits. But the `HasRoles` trait already contains `HasPermissions` so they shouldn't be used together.

**Comment by adelf** [comment]
Fixed in 4.2.1. You can delete vendor/_laravel_idea/_ide_helper_models.php manually to prevent this error.
